File: code/euler/python/ej_003.py
import math

def factors(value):
    factors = []
    factor = 2.0
    new_value = value
    val_sqrt = math.sqrt(value)

    while factor < val_sqrt:
        if new_value % factor == 0:
            factors.append(factor)
            #new_value /= factor

        factor += 1

    return factors

def calculate(value):
    return factors(value)[-1]

def print_results():
    print calculate(600851475143)

if __name__ == '__main__':
    print_results()

File: code/euler/erlang/ej_003.erl
-module(ej_003).
-compile(export_all).

factors(Factor, _Value, Limit, Accum) when Factor > Limit -> lists:reverse(Accum);
factors(Factor, Value, Limit, Accum) when Value rem Factor == 0 ->
  factors(Factor + 1, trunc(Value/Factor), Limit, [Factor | Accum]);
factors(Factor, Value, Limit, Accum) ->
  factors(Factor + 1, Value, Limit, Accum).

factors(Value) -> factors(2, Value, trunc(math:sqrt(Value)), []).

print_results() ->
  Factors = factors(600851475143),
  io:format("~w~n", [Factors]),
  if
    length(Factors) > 0 -> io:format("~w~n", ([lists:last(Factors)]));
    true -> io:format("No factor~n")
  end.

